ykbh/src/lib.rs

```diff
@@ -35,12 +35,13 @@ impl SIRInterpreter {
     }
 
     /// Sets the current frame of the interpreter.
-    pub fn set_frame(&mut self, tio: usize) {
+    pub fn set_frame(&mut self, tio: *mut usize) {
```

Comment:
Can you tell me what the model of the "current frame" is here, and what "tio" is? The description "sets the current frame" makes sense, but I can't relate that to the code inside this function. That's probably my bad, but I'm going to need your help to make sense of this!

Reply:
At the moment this is pretty much just a stub, since we haven't implemented the relevant part in the SIR yet, which will pass in the `tio` pointer (which points to the same trace inputs used by the yk compiler) and other initialised variables. The goal of this function is to initialise the current frame of the SIR we are going to interpret, which includes setting the program counter to the next instruction to be executed. Maybe I can add the program counter bit in this PR to make more clear what this does?

Reply:
OK, so this isn't really about "setting the frame" then? `set_trace_inputs` or something?

Reply:
At the moment, no. But soon it will be. So we have two options to fix this:
1) We rename it, and name it back once it actually sets the frame.
2) I add initialisation of the program counter to this function, to make clearer that this will be setting the frame soon.

Reply:
Let's name and document it such that it makes sense in this PR. Future PRs can adjust it as necessary.

Reply:
Fixed in 801159b.